src/ripple/basics/impl/IOUAmount.cpp

```diff
@@ -94,154 +117,46 @@ IOUAmount::operator+=(IOUAmount const& other)
         return *this;
     }
 
-    auto m = other.mantissa_;
-    auto e = other.exponent_;
-
-    while (exponent_ < e)
-    {
-        mantissa_ /= 10;
-        ++exponent_;
-    }
-
-    while (e < exponent_)
+    if (*stNumberSwitchover)
     {
-        m /= 10;
-        ++e;
+        *this = IOUAmount{Number{*this} + Number{other}};
```

Comment:
I find the code easier to read if we add a `return *this;` here and remove the else. Also fine as-is, it's just a preference.

Reply:
We can mark this resolved. Thanks!